### PR TITLE
Implement extractErrorString

### PR DESCRIPTION
**Pull request by somewhatabstract**
# Summary

This implements the `extractErrorString` function which is a direct copy of the [`extractErrorInfo` method in react-render-server](https://github.com/Khan/react-render-server/blob/5709c900ef9a7b3fab36e354884f9b1ba98adaff/src/logging.js#L86-L104).

The difference here is that this is fully tested and has slightly stronger flow typing.

## Issues

Part of WEB-1848.

### CODE REVIEW

File: src/shared/extract-error-string.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        return error;
+    }
+
+    if (error.response && error.response.error) {
+        return `${error.response.error}: ${error.stack || ""}`;
+    }
+
+    if (error.error && error !== error.error) {
+        return extractErrorString(error.error);
+    }
```
Comment (kevinbarabash) [inline src/shared/extract-error-string.js]: I didn't realize errors could be recursive in nature.

File: src/shared/__tests__/extract-error-string.test.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        // Assert
+        expect(result).toBe("THIS IS THE STACK!");
+    });
+
+    it("should return the result of toString if none of the other cases match", () => {
+        // Arrange
+        const error = {
+            this: "IS NOT LIKE THE OTHERS",
+            toString: () => "AND SO THIS IS THE ERROR!",
```
Comment (kevinbarabash) [inline src/shared/__tests__/extract-error-string.test.js]: Using `this` as a key here isn't necessary is it?  It's a joke right?

Comment (somewhatabstract) [reply src/shared/__tests__/extract-error-string.test.js]: Yeah, it's just a joke :grin:

File: src/shared/extract-error-string.js (hunk 1 above)
Comment (somewhatabstract) [reply src/shared/extract-error-string.js]: Yeah, this was an exciting discovery as I tried to get RRS to output useful error info.

File: src/shared/__tests__/extract-error-string.test.js (hunk 2 above)
Comment (kevinbarabash) [reply src/shared/__tests__/extract-error-string.test.js]: It was funny, but seeing `this` as a key got my spidey sense tingling.  Please don't take the fact that I asked if it was a joke as it being unfunny. 😅

Comment (somewhatabstract) [reply src/shared/__tests__/extract-error-string.test.js]: Can you imagine typing `this.this`? Imagine reviewing that code. lol

**Pull request merged**